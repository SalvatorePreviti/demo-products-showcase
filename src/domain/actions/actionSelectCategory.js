import { SELECT_CATEGORY } from '../actionTypes'

export default function actionSelectCategory(category) {
  return { type: SELECT_CATEGORY, category }
}
