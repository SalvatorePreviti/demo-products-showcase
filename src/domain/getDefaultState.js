function getDefaultState() {
  return {
    categories: [],
    activeCategory: null,
    productFilter: ''
  }
}

export default getDefaultState
