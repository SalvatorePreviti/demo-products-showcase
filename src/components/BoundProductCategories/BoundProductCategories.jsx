import React from 'react'
import { connect } from 'react-redux'
import ProductCategories from '../../atoms/ProductCategories/ProductCategories'
import actionSelectCategory from '../../domain/actions/actionSelectCategory'

function mapStateToProps(state) {
  const { categories, activeCategory } = state
  return { categories, activeCategory }
}

function mapDispatchToProps(dispatch) {
  return {
    onSelectCategory(category) {
      dispatch(actionSelectCategory(category))
    }
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(React.memo(ProductCategories))
