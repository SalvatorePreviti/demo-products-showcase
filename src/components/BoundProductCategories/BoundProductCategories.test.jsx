import React from 'react'
import getDefaultState from '../../domain/getDefaultState'
import mockStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import BoundProductCategories from './BoundProductCategories'
import actionSelectCategory from '../../domain/actions/actionSelectCategory'

describe('BoundProductCategories', () => {
  const categories = [{ id: '1', title: 'title 1' }, { id: '2', title: 'title 1' }, { id: '3', title: 'title 1' }]

  let root
  let store

  beforeAll(() => {
    store = mockStore()({
      ...getDefaultState(),
      categories: categories,
      activeCategory: categories[3]
    })
    root = shallow(<BoundProductCategories store={store} />)
  })

  it('snapshots', () => {
    expect(root).toMatchSnapshot()
  })

  it('passes props', () => {
    expect(root.props()).toMatchObject({
      categories,
      activeCategory: categories[3]
    })
  })

  it('passes callbacks', () => {
    root.simulate('selectCategory', categories[2])
    expect(store.getActions()).toEqual([actionSelectCategory(categories[2])])
  })
})
