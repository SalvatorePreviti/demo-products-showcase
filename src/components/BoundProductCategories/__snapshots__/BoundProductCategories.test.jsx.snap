// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoundProductCategories snapshots 1`] = `
<React.Memo
  categories={
    Array [
      Object {
        "id": "1",
        "title": "title 1",
      },
      Object {
        "id": "2",
        "title": "title 1",
      },
      Object {
        "id": "3",
        "title": "title 1",
      },
    ]
  }
  onSelectCategory={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
